extensions/amp-jwplayer/1.0/component.js

```diff
@@ -0,0 +1,312 @@
+import {tryParseJson} from '#core/types/object/json';
+
+import * as Preact from '#preact';
+import {useCallback, useMemo, useRef} from '#preact';
+import {forwardRef} from '#preact/compat';
+
+import {JwplayerToDom, makeJwplayerIframeSrc} from './jwplayer-api';
+
+import {isJsonOrObj, objOrParseJson} from '../../../src/iframe-video';
+import {VideoIframe} from '../../amp-video/1.0/video-iframe';
+
+const JWPLAYER_ORIGIN = /https:\/\/content\.jwplatform\.com/;
+
+/**
+ * Checks if the current browser supports requesting fullscreen
+ * @param {Window} win
+ * @return {boolean}
+ */
+function hasFullScreenApi(win) {
+  return !!win.document.body.requestFullscreen;
+}
+
+/**
+ * @param {!BentoJwplayerDef.Props} props
+ * @param {{current: ?VideoWrapperDef.Api}} ref
+ * @return {PreactDef.Renderable}
+ */
+export function BentoJwplayerWithRef(props, ref) {
+  const {
+    adCustParams,
+    adMacros,
+    autoplay,
+    config,
+    consentParams = {},
+    contentBackfill = true,
+    contentRecency,
+    contentSearch,
+    mediaId,
+    onLoad,
+    onPlayingState,
+    playerId,
+    playlistId,
+    queryParams,
+    ...rest
+  } = props;
+  const playerStateRef = useRef({});
+  const src = useMemo(() => {
+    return makeJwplayerIframeSrc(
+      playerId,
+      playlistId,
+      mediaId,
+      contentSearch,
+      contentRecency,
+      contentBackfill,
+      queryParams,
+      consentParams
+    );
+  }, [
+    playerId,
+    playlistId,
+    mediaId,
+    contentSearch,
+    contentRecency,
+    contentBackfill,
+    queryParams,
+    consentParams,
+  ]);
+
+  const onMessage = useCallback(
+    (e) =>
+      onMessage_(e, {
+        config,
+        adCustParams,
+        adMacros,
+        onPlayingState,
+        onLoad,
+        playerStateRef,
+      }),
+    [config, adCustParams, adMacros, onPlayingState, onLoad]
+  );
+
+  const makeMethodMessage = useCallback(
+    (method) => makeMessage_(method, {autoplay}),
+    [autoplay]
+  );
+  const makeFullscreenMessage = useMemo(
+    () =>
+      !hasFullScreenApi(window)
+        ? () =>
+            JSON.stringify({
+              method: 'setFullscreen',
+              optParams: true,
+            })
+        : null,
+    []
+  );
+
+  if (!checkProps(props)) {
+    return false;
+  }
+
+  return (
+    <VideoIframe
+      ref={ref}
+      playerStateRef={playerStateRef}
+      src={src}
+      makeMethodMessage={makeMethodMessage}
+      makeFullscreenMessage={makeFullscreenMessage}
+      onMessage={onMessage}
+      origin={JWPLAYER_ORIGIN}
+      controls
+      {...rest}
+    />
+  );
+}
+
+const BentoJwplayer = forwardRef(BentoJwplayerWithRef);
+BentoJwplayer.displayName = 'BentoJwPlayer'; // Make findable for tests.
+export {BentoJwplayer};
+
+/**
+ * Makes message to handle different media methods
+ * @param {string} apiMethod
+ * @param {*} options
+ * @return {string} message to post to video iframe
+ */
+function makeMessage_(apiMethod, options) {
```

Comment:
These don't need the private marker
```suggestion
function makeMessage(apiMethod, options) {
```